fish-rust/src/common.rs

```diff
@@ -89,5 +92,67 @@ pub fn escape_string(s: &wstr, style: EscapeStringStyle) -> WString {
         EscapeStringStyle::Regex => ffi::escape_string_style_t::STRING_STYLE_REGEX,
     };
 
-    ffi::escape_string(s.as_ptr(), flags_int.into(), style).from_ffi()
+    ffi::escape_string(c_str!(s), flags_int.into(), style).from_ffi()
```

Comment:
I think this is the fix referred to by the commit message.
The rest of the commit is probably largely unrelated, please split to make it easier to review